### ISSUE TITLE
SSA file loader ignores files with complex extensions

### ISSUE THREAD

**Issue by rgeronimi** [golangci/golangci-lint#158]
I am trying golangci-lint on a medium-size Go repository (excluding third party packages, 1000 files, 60000 lines of code).

It works mostly fine and is very fast. But if fails on a few very specific files whose name follow the pattern ```*.qtpl.go```. They have be generated automatically by the ```quicktemplate``` template engine.

On such files it complains that the ```goconst```and ```golint``` linters fail, because they cannot find the AST for these files in the AST cache, e.g.:
```
WARN [runner] Can't run linter goconst: no AST for file src/.../additem.qtpl.go in cache: {m:map[...]}
```

There is maybe some kind of file name pattern filter somewhere that prevents loading the AST for these files. E.g., because it is not properly parsing the extension - the dot must be searched from the right, as is the case in ```filepath.Ext```, and not from the left.

I had a very limited cursory glance in the repo to find that filter but couldn't find it.

**Comment by jirfag** [comment]
reopen if reproduces with additional information please

**Comment by rgeronimi** [comment]
Yes sorry I was in personal and professional trips. I will be able to focus on that in second half of sept. Talk to you then.

**Comment by rgeronimi** [comment]
Here it is, I isolated the minimal code and configuration that triggers the bug.

1. go version && go env
```go version go1.11.4 darwin/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/raphaelgeronimi/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/raphaelgeronimi/Local/Go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/Cellar/go/1.11.4/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.11.4/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/mk/rhxz6w7n6pbbmqt0gc2_3dg00000gn/T/go-build246439943=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

2. all errors/warnings from golangci-line
```
► golangci-lint run --enable goconst src/sample
WARN [runner] Can't run linter goconst: no AST for file /Users/raphaelgeronimi/Local/Go/src/sample/template.qtpl.go in cache: {m:map[/Users/raphaelgeronimi/Local/Go/src/sample/sample.go:0xc007813920 /Users/raphaelgeronimi/Local/Go/src/sample/template.qtpl:0xc007813950] s:[0xc007813920 0xc007813950] log:{rd:0xc0001da2a0 tags:[loader astcache] origLog:0xc00959c0c0}} 
```

3. This fails on a minimal "sample" go package with:
- A go file ensuring qtc template generation:
```
package sample

//go:generate qtc
```
- A minimal .qtpl template:
```
{% code
%}
```
automatically transformed into a .go file by the qtc tool (https://github.com/valyala/quicktemplate/tree/master/qtc)
by calling within the package
```
go generate
```

**Comment by jirfag** [comment]
Please, re check, it was fixed a few major releases ago
